### ISSUE TITLE
Returns

### ISSUE THREAD

**Issue by uRandomAlex** [danielga/gm_interstate#1]
Will RunOnClient, RunOnServer and RunOnMenu functions return result from code execution using them?

**Comment by danielga** [comment]
They either return a boolean saying the function call was successful or a nil and a error string (two values returned).

**Comment by uRandomAlex** [comment]
Can you make it return what runned code returns? I.e. this:
```
local et = RunOnMenu( "return 37" )
local le = 37
return le = et
```
will return true.

**Comment by danielga** [comment]
Not really, this implies getting the values from one Lua state and pushing them to the other. The problem isn't with strings or numbers or bools. The problem arises when you have userdata like players or weapons.

**Comment by uRandomAlex** [comment]
@danielga Look at the 'net' library extension source code: https://github.com/garrynewman/garrysmod/blob/master/garrysmod/lua/includes/extensions/net.lua
What's with tables and meta-tables?

**Comment by danielga** [comment]
Oh yeah, that's another problem, I can't just magically push tables around. Have to manually rebuild them. What a mess.

**Comment by uRandomAlex** [comment]
@danielga Weapons are entities too, so trick that "net.Write/ReadEntity" using should work. I'm sure we can transfer every special data type using Net extension example. Can you make real returns based on this anytime soon?

**Comment by danielga** [comment]
Anytime soon, no way.

**Comment by uRandomAlex** [comment]
@danielga I've found this when I was googling: https://github.com/Leystryku/Synchronicity/blob/master/src/bitbuf.cpp
I'm surprised that things like this exist, but I think this should help you.

**Comment by danielga** [comment]
It won't. My problem is not handling the stuff on the module, it's how to make Lua interact with the data I feed it. Because the menu state is so different from all the others, I don't see how I can do this at all. Entities don't exist on the menu state. It's just so much work for nothing to be honest. I can see adding support for basic data (strings, numbers, bools, etc) but not for userdata.

**Comment by uRandomAlex** [comment]
Basic data support is great too.
